### DISCUSSION TITLE
tokens are not being stored on redis db

### DISCUSSION THREAD

**Discussion by MajnuRangeela** [spencerwooo/onedrive-vercel-index#367] [Q&A]
I've messaged in your channel discussion. Store Tokens option is keep on running & as you've told me earlier to delete stored tokens from redis I came to know that redis db is getting access req from index but it doesn't write any data to db which is strange because i tried to store data from other source data is stored comfortably. So I didn't get it what is the issue..

I'm attaching some related screenshots incase it may help.

![image](https://user-images.githubusercontent.com/84721324/152768621-ba5dbcad-3c0e-47a9-b91b-091d50661ccf.png)

![image](https://user-images.githubusercontent.com/84721324/152769816-361ff5a4-5f1e-4826-a40f-25b1a6bfe195.png)

As you can see in this graph only commands are there only one write occurred which I manually did it.

**Comment by spencerwooo** [comment]
Check your redis url, what does it look like? Your issue may be similar to https://github.com/spencerwooo/onedrive-vercel-index/discussions/362

**Reply by MajnuRangeela** [reply]
my redis db url:
rediss://:b2abcxxxxx46da99d6xxxxxdb38@usw1-upward-lizard-32641.upstash.io:XXXXX

**Reply by spencerwooo** [reply]
If you enabled SSL for your Redis database, I think the url you provided looks ok. Check your Vercel function logs when you are storing the tokens.

**Reply by MajnuRangeela** [reply]
Realtime logs:

[GET] /_next/data/lWN9fWalcBlWAagJXydGV/onedrive-vercel-index-oauth/step-3.json?authCode=0.AXEAhUQpsSWULUSnd3BQvtybtu77UhMcpoBHmioS4k1wOPhxACA.AQABAAIAAAD--DLA3VO7QrddgJg7Wevrg7_fdPQA1zsYBc9DveLZ3ceeAziCcxt7TEjP-LmuS_rbpeqCV76N_F-UaZBS-UlijMNMOdoVpFlrspMWQ49BOp5SWkUhrcM-L72OLEHWFUL80yqgTAI2kMlEkqlLH8xgrhzz-LFjnq0ztCeUksM6EBVwYD686zynBqB0IWPTF-0iyEg8yEMfx2V2rPBXgVbrMB98sYmkHuhjgTPP6W9HrLQLCTxIHYjVNtgHF7dU0gHnipDlVfxzLQUqjypYC2TU-u1TLRXylUBXFmk6KSWWXyeTLb92tPcZ3HaAxoRG4NgSrPLqaDyb2N3SPWxBn3oDl9LpfJMpUU52lfDBb3xoRYiKdSND2xx5if0MaZ3zIXGycXP3xY7wduWEzBx9p4tR3M9u_DFWD_74SbTewQKpTKJt9EZTQnTzZUjB4-ro2IpO_FfDZ9UjvaiOR4rwxei0r-Z-k8q30GaPysGZvIxTBB5bHWpHa3X3RvDd2Jp6YgZtWDJ-riI93i3txp18XPIQp79xmehakS7ZaKNHG2THKVf2cJddrTbLDT_KKsRPamfsTpIS0bfYM2JBP5Vbkexb3kYvormvNWA8c98nLpwLNCAA
16:57:31:34
Function Status:
200
Edge Status:
200
Duration:
563.56 ms
Init Duration:
N/A
Memory Used:
103 MB
ID:
bom1::iad1::r595p-1644233251343-7dedf99d041a
User Agent:
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Safari/537.36
[GET] /api?path=/
16:56:55:52
Function Status:
403
Edge Status:
403
Duration:
128.69 ms
Init Duration:
N/A
Memory Used:
97 MB
ID:
bom1::iad1::r595p-1644233215525-695758251e14
User Agent:
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.87 Safari/537.36
No refresh token, return empty access token.

& No errors detected.
